### DISCUSSION TITLE
How is the Card ID derived?

### DISCUSSION THREAD

**Discussion by cristtos** [MiczFlor/RPi-Jukebox-RFID#2308] [Q&A]
I have tried to find here some clarification but could not find anything relevant - or used the wrong keywords.

We bought a bunch of no name cards.
When scanned by my Galaxy each scans with a different Serial Numbers but the web interface shows only 2 different Card IDs.
So I have 30 cards that read with one ID and 20 that read with the other ID.
We have also bought 10 sticky tags and all 10 read with a singe ID (different from the above).

We ended up with some 60 tags from which we can issue 3 commands.

Can someone explain how the Card ID is derived and if it is possible to get these cards to function with independent IDs?
Or maybe we are doing something wrong?

The software works fine with these 3 cards.

Thanks 

Cris

**Comment by AlvinSchiller** [comment]
Hi, sounds strange indeed. 
Which version do you have installed and what card reader do you use?

**Reply by cristtos** [reply]
Thank you for answering my query.
The Version is 2.5.0 - 4101984b - master.
The reader is a USB from TEMU, Link to the product is not available anymore but the device is detected as Realtek USB2,0 Card Reader on my windows laptop.
It came with some brief "instructions", in the PNG attached.
![CardReader](https://github.com/MiczFlor/RPi-Jukebox-RFID/assets/9064557/a8e96d28-ffc4-4d27-93af-08005ec49097)

This is rather confusing...

Thanks again 

Cris

**Reply by cristtos** [reply]
While playing with the reader on my laptop I figured out that it actually "types" the Card ID in a Notepad window.
It types same thing:
1745944969
1745944969
when tapped with 2 different cards, cards that show different serials on my Galaxy NFC tools.

Not sure if this helps.

**Reply by s-martin** [reply]
Are your cards and your reader NFC or RFID?
Maybe you are mixing an NFC reader with RFID cards or vice versa.

While there's some overlap, they are not the same.

Usually we use RFID cards and readers.
Protocols to look for the cards are e.g. EM4100 (125 kHz) or Mifare (13.56 MHz). Select a reader and then matching cards.

Simplest option is the Neuftech USB RFID Reader, which uses EM4100.

Others are RC522 reader (which needs soldering), which uses Mifare.

**Reply by s-martin** [reply]
See also here https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/future3%2Fmain/documentation%2Fdevelopers%2Frfid%2FREADME.md

**Reply by cristtos** [reply]
Many thanks for the pointers, will do some reading.
I installed the NFC tools on the Windows laptop but it does not detect any NFC readers.
I think the USB reader works as a simple fake keyboard, as it types in my notepad.
It also can "read" my credit cards - but I am not sure if this means anything.
The phone is definitely NFC.

I will also try to install the newest build on a different SD card.

Thanks again

Cris
